ProcessMaker/Http/Controllers/Api/ProcessController.php

```diff
@@ -54,18 +59,39 @@ class ProcessController extends Controller
      */
     public function index(Request $request)
     {
-        $where = $this->getRequestFilterBy($request, ['processes.name', 'processes.description','processes.status', 'category.name', 'user.firstname', 'user.lastname']);
+        $where = $this->getRequestFilterBy($request, ['processes.name', 'processes.description', 'processes.status', 'category.name', 'user.firstname', 'user.lastname']);
         $orderBy = $this->getRequestSortBy($request, 'name');
         $perPage = $this->getPerPage($request);
         $include = $this->getRequestInclude($request);
         $processes = Process::with($include)
             ->select('processes.*')
-            ->where($where)
             ->leftJoin('process_categories as category', 'processes.process_category_id', '=', 'category.id')
             ->leftJoin('users as user', 'processes.user_id', '=', 'user.id')
-            ->orderBy(...$orderBy)
-            ->paginate($perPage);
-        return new ApiCollection($processes);
+            ->where($where)
+            ->orderBy(...$orderBy);
+
+        if (!Auth::user()->is_administrator) {
+            $startPermission = Permission::byGuardName('requests.create')->id;
+            $processes->whereExists(function($query) use ($startPermission){
+                $query->select(\DB::raw(1))
+                    ->from((new ProcessPermission)->getTable())
+                    ->whereRaw("process_permissions.process_id = processes.id " .
+                                " AND permission_id=" . $startPermission  .
+                                " AND assignable_type = '" . str_replace('\\', '\\\\', User::class) . "'" .
+                                " AND assignable_id = " . Auth::user()->id);
+            });
+
+            $processes->orWhereExists(function($query) use ($startPermission){
+                $query->select(\DB::raw(1))
+                    ->from((new ProcessPermission)->getTable())
+                    ->whereRaw("process_permissions.process_id = processes.id " .
+                        " AND permission_id=" . $startPermission  .
+                        " AND assignable_type = '" . str_replace('\\', '\\\\', Group::class) . "'" .
+                        " AND assignable_id IN (select group_id from group_members where member_id = " .
+                        Auth::user() ->id . " )");
```

Comment:
I think there's probably a cleaner way to do this. We should try to avoid any raw sql when ever possible. I think the logic should get moved out of the controller and into the [chainable scope](https://laravel.com/docs/5.7/eloquent#local-scopes) on the process model, something like `forUser($user)`